x-pack/plugins/apm/server/routes/service_map/get_service_map_service_node_info.ts

```diff
@@ -172,15 +202,36 @@ async function getTransactionStats({
   const totalRequests = response.hits.total.value;
 
   return {
-    avgTransactionDuration: response.aggregations?.duration.value ?? null,
-    avgRequestsPerMinute: totalRequests > 0 ? totalRequests / minutes : null,
+    latency: {
+      value: response.aggregations?.duration.value ?? null,
+      timeseries: response.aggregations?.timeseries.buckets.map((bucket) => ({
+        x: bucket.key,
+        y: bucket.latency.value,
+      })),
+    },
+    throughput: {
+      value: totalRequests > 0 ? totalRequests / minutes : null,
+      timeseries: response.aggregations?.timeseries.buckets.map((bucket) => {
+        return {
+          x: bucket.key,
+          y: calculateThroughputWithRange({
+            start,
+            end,
+            value: bucket.doc_count ?? 0,
+          }),
```

Comment:
Is this throughput correct? It looks like you are using the value from the bucket but the overall time window. Might not make a big difference since this is just for sparklines that don't have values on axis but it's a bug waiting to become visible.

To circumvent this, can you use rate agg instead of manually calculating throughput?

Or alternatively you can disregard the rate and just use doc_count:

```suggestion
          y: bucket.doc_count ?? 0,
```